src/mapper/pkg/dnsintentspublisher/dns_intents_publisher.go

```diff
@@ -163,19 +164,26 @@ func (p *Publisher) compareIntentsAndStatus(clientIntents otterizev2alpha1.Clien
 }
 
 func (p *Publisher) appendResolvedIps(dnsName string, resolvedIPsMap map[string]map[string]struct{}) bool {
-	resolvedIPs := p.dnsCache.GetResolvedIPs(dnsName)
+	resolvedIPs := make([]string, 0)
+	if p.isWildcardDNS(dnsName) {
+		matchingDNSNames := p.dnsCache.GetMatchingEntriesForWildcard(dnsName)
+		for _, name := range matchingDNSNames {
+			resolvedIPs = append(resolvedIPs, p.dnsCache.GetResolvedIPs(name)...)
+		}
+	} else {
+		resolvedIPs = p.dnsCache.GetResolvedIPs(dnsName)
+	}
 
 	ips, ok := resolvedIPsMap[dnsName]
 	if !ok {
 		ips = make(map[string]struct{})
 	}
 
 	if len(resolvedIPs) == 0 {
```

Comment:
you still need to check if !isWildcardDNS (and avoid trying to resolve it ourselves). 
it could be that there's 0 matching DNS names, in which case you'll get here with an empty list of resolved IPs.